src/query.js

```diff
@@ -59,18 +55,23 @@ const encodeMaps = (maps, config) => {
     let encodeMap = { id: m.id };
     let configMap;
 
-    if (m.branchPatternId) {
-      configMap = findBranchPattern(m, config.branchPatterns);
+    if (m.branchId) {
+      configMap = findBranchPattern(m, config?.branchPatterns ?? []);
       encodeMap = {
-        branchPatternId: configMap.id,
-        branchPatternStyle: m.branchPatternStyle,
+        branchId: configMap.id,
+        branchStyle: m.branchStyle,
         branch: m.branch,
       };
     } else {
       configMap = findStylePreset(m.id, config.stylePresets);
     }
 
-    if (!configMap) return m;
+    // If not set yet, must be a custom URL
+    if (!configMap) {
+      return Object.fromEntries(
+        Object.entries(m).filter(([k]) => k !== 'style')
```

Comment:
Technically I think we could get by with even even fewer properties for a custom URL, but I think you're right that it's better to be safe here and lump it in with any other edge case.